dash/dash.py

```diff
@@ -149,12 +208,26 @@ def layout(self, value):
         # pylint: disable=protected-access
         self.css._update_layout(layout_value)
         self.scripts._update_layout(layout_value)
-        self._collect_and_register_resources(
-            self.scripts.get_all_scripts()
-        )
-        self._collect_and_register_resources(
-            self.css.get_all_css()
+
+    @property
+    def index_string(self):
+        return self._index_string
+
+    @index_string.setter
+    def index_string(self, value):
+        checks = (
+            (_re_index_entry.search(value), 'app_entry'),
+            (_re_index_config.search(value), 'config',),
+            (_re_index_scripts.search(value), 'scripts'),
         )
+        missing = [missing for check, missing in checks if not check]
+        if missing:
+            raise Exception(
```

Comment:
This should be a Dash-specific exception from the `dash.exceptions` module, but there's already a couple of other plain `Exception`s in `dash.py`, so it could just be fixed in another PR.